props/saber_fett263_buttons.h

```diff
@@ -285,47 +440,1090 @@ EFFECT(faston); // for EFFECT_FAST_ON
+#else
+    CONFIG_VARIABLE(stabon, 0);
+#endif
+#ifdef FETT263_TWIST_OFF
+    CONFIG_VARIABLE(twistoff, 1);
+#else
+    CONFIG_VARIABLE(twistoff, 0);
+#endif
+#ifdef FETT263_POWER_LOCK
+    CONFIG_VARIABLE(powerlock, 1);
+#else
+    CONFIG_VARIABLE(powerlock, 0);
+#endif
+#ifdef FORCE_PUSH_CONDITION
+    CONFIG_VARIABLE(forcepush, 1);
+#else
+    CONFIG_VARIABLE(forcepush, 0);
+#endif
+    CONFIG_VARIABLE(forcepushlen, FETT263_FORCE_PUSH_LENGTH);
+    CONFIG_VARIABLE(lockupdelay, FETT263_LOCKUP_DELAY);
+    CONFIG_VARIABLE(clashdetect, FETT263_BM_CLASH_DETECT);
+#ifdef FETT263_MAX_CLASH
+    CONFIG_VARIABLE(maxclash, FETT263_MAX_CLASH);
+#else
+    CONFIG_VARIABLE(maxclash, 16);
+#endif
+
+}
+  bool gestureon;
+  bool swingon;
+  int swingonspeed;
+  bool twiston;
+  bool thruston;
+  bool stabon;
+  bool forcepush;
+  int forcepushlen;
+  int lockupdelay;
+  bool twistoff;
+  bool powerlock;
+  int clashdetect;
+  int maxclash;
+
+};
+
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+// Rehearsal / Choreography
+class SavedRehearsal : public ConfigFile {
+public:
+  virtual ReadStatus Read(FileReader* f) override {
+    int file;
+    if (!f || !f->IsOpen()) return ReadStatus::READ_FAIL;
+    memset(clash_rec, 0, sizeof(clash_rec));
+    STDOUT.println("Reading choreo.ini");
+    for (size_t i = 0; i < NELEM(clash_rec); i++) {
+      char action[33];
+      f->skipwhite();
+      f->readVariable(action);
+      if (!strcmp(action, "clash")) {
+          if (SFX_clsh) {
+             file = SFX_clsh.get_min_file();
+          } else {
+             file = SFX_clash.get_min_file();
+          }
+        clash_rec[i].stance = STANCE_CLASH;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        continue;
+      }
+      if (!strcmp(action, "lockup")) {
+          if (SFX_lock) {
+            file = SFX_lock.get_min_file();
+          } else {
+            file = SFX_lockup.get_min_file();
+          }
+        clash_rec[i].stance = STANCE_LOCKUP;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - SFX_bgnlock.get_min_file());
+        f->skipspace();
+        clash_rec[i].sound_number2 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        clash_rec[i].sound_number3 = (int)(f->readFloatValue() - SFX_endlock.get_min_file());
+        f->skipspace();
+        continue;
+      }
+      clash_rec[i].stance = STANCE_END;
+      return ReadStatus:: READ_END;
+      break;
+    }
+  }
+
+  void SetVariable(const char* variable, float v) override {
+  }
+
+  enum FormStance {
+    STANCE_END = 0,
+    STANCE_CLASH,
+    STANCE_LOCKUP,
+  };
+
+  struct FormEntry {
+    FormStance stance;
+    int sound_number1;
+    int sound_number2;
+    int sound_number3;
+  };
+
+  FormEntry clash_rec[32];
+ 
+};
+#endif
+
+#ifdef FETT263_EDIT_MODE_MENU
+// Edit Length 
+// NewLength gets length value from menu for use in LengthEdit
+class NewLength {
```

Comment:
I don't think we need this class.
A simple global variable will be sufficient, like:  int length_edit_length = 0;

Reply:
updated, thanks.